shared/constants/teams.js

```diff
@@ -227,6 +227,12 @@ const getSelectedTeamNames = (state: TypedState): Types.Teamname[] => {
   }, [])
 }
 
+/**
+ *  Gets the number of channels you're subscribed to on a team
+ */
+const getNumberOfSubscribedChannels = (state: TypedState, teamname: Types.Teamname): number =>
+  state.chat2.metaMap.filter(c => c.teamname === teamname).count()
```

Comment:
Ooh didn't know about count, it looks like it can take a predicate [directly](https://facebook.github.io/immutable-js/docs/#/Map/count) so you can get rid of the `filter` here